internal/pkg/policy/monitor.go

```diff
@@ -128,6 +135,35 @@ LOOP:
 	return nil
 }
 
+// Aggegates changes from the output channel until the main loop can process.
+func (m *monitorT) monitorOutputChannel(ctx context.Context, outputCh <-chan []es.HitT) chan []es.HitT {
+	localOutputCh := make(chan []es.HitT)
+
+	go func() {
+
+		var hits []es.HitT
+		var outCh chan []es.HitT
+
+		for {
+			select {
+			case <-ctx.Done():
+				m.log.Info().Msg("Exit policy monitor local")
+				return
+			case nHits := <-outputCh:
+				m.log.Info().Int("nHits", len(nHits)).Msg("Received hits on local monitor")
+				hits = append(hits, nHits...)
+				outCh = localOutputCh
```

Comment:
Why is `outCh` set in this case? Couldn't it be set at the top of the go func or just use `localOutputCh` directly. Or am I missing something about this flow.

Reply:
The localOutputCh  has no buffer by design.  If we use it directly it will block and we won't pickup any additional changes on outputCh.

We could make localOutpuCh a buffered channel, but how big is the buffer?  You may still hit the max size and block.  

This method allows us to continue to add data to the 'hits' array until the consuming loop is able to retrieve the data.  Switching 'outCh' to null once consumed allows the loop to spin properly until additional data is available.

Reply:
Thanks for the explanation. I missed the `outCh = nil` in the last case, that make sense now.